tee-worker/litentry/core/assertion-build/src/dynamic/contracts/token_holding_amount/TokenMapping.sol

```diff
@@ -396,6 +406,8 @@ contract TokenMapping is TokenQueryLogic {
 		tokenAddresses["usdc"][Web3Networks.Bsc] = Usdc.getTokenBscAddress();
 		tokenAddresses["usdc"][Web3Networks.Ethereum] = Usdc
 			.getTokenEthereumAddress();
+		tokenAddresses["usdc"][Web3Networks.Solana] = Usdc
+			.getTokenSolanaAddress();
```

Comment:
2) I think we should provide a single function with network param to get the token address. e.g.
-- Dai.sol
function getTokenAddress(uint32 network) internal pure returns (string memory) {
  ...
}

function getTokenNetworks() internal pure returns (uint32[] memory) {
  ...
}

-- TokenMapping.sol

constructor() {
  ...
  uint32[] memory networks = Dai.getTokenNetworks();
  for (uint i = 0; i < networks.length; i++) {
    tokenAddresses["dai"][networks[i]] = Dai.getTokenAddress(networks[i]);
  }
  ...
}

Reply:
I simply tested the modified results, evm:bnb and solana:mcrt, both passed the test:
```./bin/litentry-cli trusted -d request-vc did:litentry:substrate:0xe659a7a1628cdd93febc04a4e0646ea20e9f5f0ce097d9a05290d4a9e054df4e -a "dynamic 0000000000000000000000000000000000000002 000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000
000000000000000000000000000000000000046d63727400000000000000000000000000000000000000000000000000000000 true"
>>> identity: Substrate(Address32)
>>> nonce: 2
>>> assertions: [Dynamic(DynamicParams { smart_contract_id: 0x0000000000000000000000000000000000000002, smart_contract_params: Some(BoundedVec([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 32, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 109, 99, 114, 116, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 1024)), return_log: true })]
----Generated VC-----
{"@context":["https://www.w3.org/2018/credentials/v1","https://w3id.org/security/suites/ed25519-2020/v1"],"id":"0x14c4dd611da4fc1e1ca70907e0896ca9cbb73b63672ffb237865df0876992a17","type":["VerifiableCredential"],"credentialSubject":{"id":"did:litentry:substrate:0xe659a7a1628cdd93febc04a4e0646ea20e9f5f0ce097d9a05290d4a9e054df4e","description":"The amount of a particular token you are holding","type":"Token Holding Amount","assertionText":"","assertions":[{"and":[{"src":"$token","op":"==","dst":"mcrt"},{"src":"$holding_amount","op":">=","dst":"50000"},{"src":"$holding_amount","op":"<","dst":"150000"}]}],"values":[true],"endpoint":"http://localhost:9933"},"issuer":{"id":"did:litentry:substrate:0xdf45100d260ed1e7e0d53c400858773d1263e814289f8f250499b6fa2a5d01c9","name":"Litentry TEE Worker","mrenclave":"DPtRXAxntVPL49TyARdwfJgHYHe23QtZC1bxV18wnLnd","runtimeVersion":{"parachain":9181,"sidechain":108}},"issuanceDate":"2024-07-18T08:39:57.126861771+00:00","parachainBlockNumber":135,"sidechainBlockNumber":264,"proof":{"created":"2024-07-18T08:39:57.127067942+00:00","type":"Ed25519Signature2020","proofPurpose":"assertionMethod","proofValue":"218bd134a8d10635fe92c91d3a5cf6285a3a7a443666559242e68a9d197f19dac91db45679b9fea9d9ff3d4cde58e5ca2a023acd1e30508b810ba987caa6a106","verificationMethod":"0x6813b486dc599656b6112fd002351d3aea23162a697e6826e92d75ac834aad38"},"credentialSchema":{"id":"https://raw.githubusercontent.com/litentry/vc-jsonschema/main/dist/schemas/25-token-holding-amount/1-1-0.json","type":"JsonSchemaValidator2018"}}
```
```
 ./bin/litentry-cli trusted -d request-vc did:litentry:substrate:0xe659a7a1628cdd93febc04a4e0646ea20e9f5f0ce097d9a05290d4a9e054df4e -a "dynamic 0000000000000000000000000000000000000001 0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000
0000000000000000000000000000000000003626e620000000000000000000000000000000000000000000000000000000000 true"
>>> identity: Substrate(Address32)
>>> nonce: 1
>>> assertions: [Dynamic(DynamicParams { smart_contract_id: 0x0000000000000000000000000000000000000001, smart_contract_params: Some(BoundedVec([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 32, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3, 98, 110, 98, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 1024)), return_log: true })]
----Generated VC-----
{"@context":["https://www.w3.org/2018/credentials/v1","https://w3id.org/security/suites/ed25519-2020/v1"],"id":"0x81b43214cdabf50c41f95fbf6ad7d9afbd7f898dab1393cde21c1a314da7b504","type":["VerifiableCredential"],"credentialSubject":{"id":"did:litentry:substrate:0xe659a7a1628cdd93febc04a4e0646ea20e9f5f0ce097d9a05290d4a9e054df4e","description":"The amount of a particular token you are holding","type":"Token Holding Amount","assertionText":"","assertions":[{"and":[{"src":"$token","op":"==","dst":"bnb"},{"src":"$holding_amount","op":">=","dst":"1"},{"src":"$holding_amount","op":"<","dst":"50"}]}],"values":[true],"endpoint":"http://localhost:9933"},"issuer":{"id":"did:litentry:substrate:0xdf45100d260ed1e7e0d53c400858773d1263e814289f8f250499b6fa2a5d01c9","name":"Litentry TEE Worker","mrenclave":"DPtRXAxntVPL49TyARdwfJgHYHe23QtZC1bxV18wnLnd","runtimeVersion":{"parachain":9181,"sidechain":108}},"issuanceDate":"2024-07-18T08:36:53.726069023+00:00","parachainBlockNumber":120,"sidechainBlockNumber":234,"proof":{"created":"2024-07-18T08:36:53.726891687+00:00","type":"Ed25519Signature2020","proofPurpose":"assertionMethod","proofValue":"77c473d2b36bc73073c02abaf373e223a774d4a2a0d081b3e4bf8e6ac4f6c285fc0550e507a0726180fd0657527f84bce89e269fcbeec3e1b01c030c0fe7e00b","verificationMethod":"0x6813b486dc599656b6112fd002351d3aea23162a697e6826e92d75ac834aad38"},"credentialSchema":{"id":"https://raw.githubusercontent.com/litentry/vc-jsonschema/main/dist/schemas/25-token-holding-amount/1-1-0.json","type":"JsonSchemaValidator2018"}}
```